### PR TITLE
fix binlogShouldBeFetched

### PR DESCRIPTION
**Pull request by g0djan**

### CODE REVIEW

File: internal/databases/mysql/stream_fetch_handler.go [hunk 1]
```diff
@@ -45,9 +45,20 @@ func downloadAndDecompressStream(folder storage.Folder, fileName string) error {
 	if err != nil {
 		return err
 	}
+	binlogs, _, err := folder.GetSubFolder(BinlogPath).ListFolder()
+	if err != nil {
+		return err
+	}
+	var backupStartUploadTime time.Time
```
Comment (mialinx) [inline internal/databases/mysql/stream_fetch_handler.go]: This name is confusing, as it's binlog uploading datetime, and it happens after backup started, and even after backup finished.

File: internal/databases/mysql/mysql_test.go [hunk 2]
```diff
@@ -7,8 +7,10 @@ import (
 	"github.com/wal-g/wal-g/internal"
 	"github.com/wal-g/wal-g/internal/databases/mysql"
 	"github.com/wal-g/wal-g/internal/storages/memory"
+	storage2 "github.com/wal-g/wal-g/internal/storages/storage"
```
Comment (mialinx) [inline internal/databases/mysql/mysql_test.go]: it's better to rename local variables not packages

File: internal/databases/mysql/stream_fetch_handler.go [hunk 3]
```diff
@@ -130,8 +141,9 @@ func fetchBinlogs(folder storage.Folder, sentinel StreamSentinelDto, binlogsAreD
 	binlogsAreDone <- nil
 }
 
-func BinlogShouldBeFetched(sentinel StreamSentinelDto, binlogName string, endTS *time.Time, object storage.Object) bool {
-	return sentinel.BinLogStart <= binlogName && (endTS == nil || (*endTS).After(object.GetLastModified()))
+func BinlogShouldBeFetched(backupStartUploadTime time.Time, endTS *time.Time, object storage.Object) bool {
+	return (backupStartUploadTime.Before(object.GetLastModified()) || backupStartUploadTime.Equal(object.GetLastModified())) &&
+		(endTS == nil || (*endTS).After(object.GetLastModified()))
```
Comment (mialinx) [inline internal/databases/mysql/stream_fetch_handler.go]: Suppose we have binlog `my-bin.0042` started at 12:00, finished at 14:00 and uploaded to S3 at 14:05
User ask for PITR at 13:00 - so we need to fetch this binlog, even if it's upload date after `endTS`

Comment (g0djan) [reply internal/databases/mysql/stream_fetch_handler.go]: It will be fixed in another PR.

**Pull request merged**